src/plugins/cams/client.py

```
@@ -0,0 +1,236 @@
+            lead_times = [0, 3, 6, 9, 12]  # Up to 12 hours ahead
+            
+        if variables is None:
+            variables = ['particulate_matter_2.5um', 'particulate_matter_10um']
+            
+        if area is None:
+            area = self.japan_area
+            
+        # Ensure base_time is 00:00 or 12:00 UTC
+        if base_time.hour not in [0, 12]:
+            base_time = base_time.replace(hour=0 if base_time.hour < 12 else 12)
+            logger.warning(f"Adjusted base time to {base_time}")
```

Ruff (0.12.2)

Logging statement uses f-string

(G004)